src/ol/interaction/Draw.js

```diff
@@ -903,9 +902,44 @@ class Draw extends PointerInteraction {
   }
 
   /**
-   * Extend an existing geometry by adding additional points. This only works
-   * on features with `LineString` geometries, where the interaction will
-   * extend lines by adding points to the end of the coordinates array.
+   * Append coordinates to the end of the geometry that is currently being drawn.
+   * This can be used when drawing LineStrings or Polygons. Coordinates will
+   * either be appended to the current LineString or the outer ring of the current
+   * Polygon.
+   * @param {!LineCoordType} coordinateExtension Linear coordinates to be appended into
+   * the coordinate array.
+   * @api
+   */
+  appendCoordinates(coordinateExtension) {
+    const mode = this.mode_;
+    let coordinates = [];
+    if (mode === Mode.LINE_STRING) {
+      coordinates = /** @type {LineCoordType} */ this.sketchCoords_;
+    } else if (mode === Mode.POLYGON) {
+      coordinates = this.sketchCoords_ && this.sketchCoords_.length ? /** @type {PolyCoordType} */ (this.sketchCoords_)[0] : [];
+    }
+
+    // Remove last coordinate from sketch drawing (this coordinate follows cursor position)
+    const ending = coordinates.pop();
+
+    // Append coordinate list
+    for (let i = 0; i < coordinateExtension.length; i++) {
+      this.addToDrawing_(coordinateExtension[i]);
+    }
+
+    // Duplicate last coordinate for sketch drawing
+    this.addToDrawing_(ending);
+  }
+
+  /**
+   * Initiate draw mode by starting from an existing geometry which will
+   * receive new additional points. This only works on features with
+   * `LineString` geometries, where the interaction will extend lines by adding
+   * points to the end of the coordinates array.
+   * This will change the original feature, instead of drawing a copy.
+   *
+   * The function will dispatch a `drawstart` event.
+   *
```

Comment:
I'm wondering why this is limited to linestrings. Shouldn't this work the same way for polygons?

Reply:
Actually I see now that what you added as `appendCoordinates` api function is more or less a replacement for this old `extend`  function. It also looks like this function is not used in any example, and looking at the new example I don't think it is very useful Couldn't this be marked `@deprecated` and removed in v7?

Reply:
For reference: `Draw#extend` was introduced in #4044. It still might be useful. @jahow, do you think you can quickly check if this works for polygons out of the box, and update the docs if this is the case?

Reply:
The `extend` method gives errors if used on polygons, or in polygon drawing mode. Even when extending an existing line, there is an error when finishing the draw. I'll create an issue with a reproduction case, maybe this API needs some more work or to be deprecated in favor of something more generic?

Anyway, the `appendCoordinates` API doesn't fill exactly the same purpose so I think both can coexist for the time being.